tests/console/oprofile.pm

```diff
@@ -0,0 +1,42 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Simple tests for oprofile.
+# - profile example application (fio to create some load)
+# - check report
+# Maintainer: Michael Grifalconi <mgrifalconi@suse.com>
+
+use strict;
+use warnings;
+use base 'consoletest';
+use testapi;
+use utils;
+
+sub run {
+    select_console 'root-console';
+
+    # Install fio as load generator and oprofile
+    zypper_call "in fio oprofile";
+
+    # Start a system wide profiling
+    type_string "operf --system-wide &";
+
+    # Start load and stop profiler when finished
+    assert_script_run "fio --name=randwrite --filename=/tmp/randwrite --bs=4k --size=1G && killall -s SIGINT operf";
+
+    # Make sure the report is generated correctly and also check for an entry for the fio load
+    assert_script_run "opreport";
+    assert_script_run "opreport | grep 'fio'";
+
+    # Cleanup
+    assert_script_run "rm -rf /tmp/randwrite oprofile_data";
```

Comment:
Not necessary here, but one could consider moving this to `post_run_hook` and `post_fail_hook`

    sub post_run_hook {
        script_run "rm -rf /tmp/randwrite oprofile_data";
    }

Reply:
Was not aware of this, makes sense! Will do.